File: src/features/tutorial-video/presenters.ts
// Single source of truth for who can appear on a tutorial's opening overlay.
// To add a presenter, add one line here — the dropdown, the render overlay, and
// the grouped history all read from this list. Nothing else needs to change.
export interface Presenter {
  id: string; // stable slug — stored/sent, never shown
  name: string; // overlay line 1 + history group key
  handle: string; // overlay line 2 (@handle)
}

export const PRESENTERS: Presenter[] = [
  { id: "rayandika", name: "Muhammad Rayandika", handle: "@rayandikacode" },
  { id: "depras", name: "Depras Nuryadi", handle: "@Deprasny" },
  { id: "rafi", name: "Muhammad Rafi Reyhan", handle: "@mrafireyhan" },
];

export const DEFAULT_PRESENTER_ID = PRESENTERS[0].id;

export function getPresenter(id: string): Presenter | undefined {
  return PRESENTERS.find((p) => p.id === id);
}

// History rows store the display name; this resolves the handle back for display
// and lets us order groups by the canonical PRESENTERS order.
export function getPresenterByName(name: string): Presenter | undefined {
  return PRESENTERS.find((p) => p.name === name);
}

File: agents/video-materials/video_materials/presenters.py
"""Mirror of the web app's presenter list.

Keep in sync with
src/features/tutorial-video/presenters.ts — same ids/names/handles so a brief's
presenter resolves identically on both sides.
"""

from __future__ import annotations

from dataclasses import dataclass


@dataclass(frozen=True)
class Presenter:
    id: str       # stable slug — stored in news_briefs.presenter_id
    name: str     # display name — news_briefs.presenter_name + thumbnail overlay
    handle: str   # @handle for the thumbnail overlay


PRESENTERS: list[Presenter] = [
    Presenter("rayandika", "Muhammad Rayandika", "@rayandikacode"),
    Presenter("depras", "Depras Nuryadi", "@Deprasny"),
    Presenter("rafi", "Muhammad Rafi Reyhan", "@mrafireyhan"),
]

DEFAULT_PRESENTER_ID = PRESENTERS[0].id


def get_presenter(presenter_id: str) -> Presenter:
    for p in PRESENTERS:
        if p.id == presenter_id:
            return p
    valid = ", ".join(p.id for p in PRESENTERS)
    raise ValueError(f"Unknown presenter '{presenter_id}'. Valid: {valid}")
